src/unix/newlib/mod.rs

```diff
@@ -232,23 +230,37 @@ s! {
 // unverified constants
 align_const! {
     pub const PTHREAD_MUTEX_INITIALIZER: pthread_mutex_t = pthread_mutex_t {
-        size: [0; __SIZEOF_PTHREAD_MUTEX_T],
+        size: [__PTHREAD_INITIALIZER_BYTE; __SIZEOF_PTHREAD_MUTEX_T],
     };
     pub const PTHREAD_COND_INITIALIZER: pthread_cond_t = pthread_cond_t {
-        size: [0; __SIZEOF_PTHREAD_COND_T],
+        size: [__PTHREAD_INITIALIZER_BYTE; __SIZEOF_PTHREAD_COND_T],
     };
     pub const PTHREAD_RWLOCK_INITIALIZER: pthread_rwlock_t = pthread_rwlock_t {
-        size: [0; __SIZEOF_PTHREAD_RWLOCK_T],
+        size: [__PTHREAD_INITIALIZER_BYTE; __SIZEOF_PTHREAD_RWLOCK_T],
```

Comment:
What is the plan here? If ESPIDF gains a pthread_rwlock implementation in the future then these numbers will likely change. I would `cfg` rwlock out until it is supported by ESPIDF.

Reply:
`pthread_mutex_t` on ESP-IDF is implemented as a pointer to a structure *allocated on the heap*. I would expect the same would hold true for `pthread_rwlock_t`, hence I kept it the same size.

These definitions might also allow me to quickly prototype an "out of tree" implementation of these in the next days/weeks and test without altering libc again.

Reply:
JFYI, [here](https://github.com/espressif/esp-idf/blob/master/components/pthread/pthread.c#L574).